tsinfer/formats.py

```diff
@@ -2480,6 +2480,127 @@ def insert_proxy_samples(
         assert other.num_ancestors == self.num_ancestors + len(sample_ids)
         return other
 
+    def truncate_ancestors(
+        self,
+        lower_time_bound,
+        upper_time_bound,
+        length_multiplier=2,
+        **kwargs,
+    ):
+        """
+        Truncates the length of ancestors above a given time and returns a new
+        :class:`.AncestorData` instance.
+
+        The maximum length of an ancestor existing between ``lower_time_bound'' and
+        ``upper_time_bound'', multiplied by ``length_multplier'', is used to truncate
+        the length of all ancestors older than ``upper_time_bound'' which contain
+        at least one ``focal_site''.
```

Comment:
Small nit here: writing ``focal_site`` like this implies that it's an input parameter or other variable. But really, it's just a "focal site", right? Same point in next para